core/services/keeper/registry_interface.go

```diff
@@ -140,10 +180,11 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 			}
 		}
 		return activeUpkeeps, nil
-	case RegistryVersion_1_2:
+	case RegistryVersion_1_2, RegistryVersion_1_3:
 		if opts == nil || opts.BlockNumber.Int64() == 0 {
+			var header *types.Header
 			// fetch the current block number so batched GetActiveUpkeepIDs calls can be performed on the same block
-			header, err := rw.evmClient.HeaderByNumber(context.Background(), nil)
+			header, err = rw.evmClient.HeaderByNumber(context.Background(), nil)
```

Comment:
We also need to pass this opt in the below getState call (which is now refactored into getUpkeepCount call)

Can you move this code block to the top of the function? Let's use the consistent block number in all eth calls

Reply:
Good catch